### ISSUE TITLE
No errors shown when having undefined components

### ISSUE THREAD

**Issue by nickjanssen** [yannickcr/eslint-plugin-react#655]
![eslintbug](https://cloud.githubusercontent.com/assets/2780160/16510726/94f95e90-3f83-11e6-95a9-3489df403e08.gif)

An error should be shown that `Text` is undefined.

To reproduce:

```
import React from 'react'

import { View } from 'react-native'

export default class Root extends React.Component {
  render() {
    return (
      <View>
        <Text>Hello!</Text>
      </View>
    )
  }
}
```

Rules:
```
{
  "extends": "plugin:react/recommended",
  "env": {
    "es6": true,
    "browser": true,
    "node": true
  },
  "plugins": [
    "react",
    "babel"
  ],
  "parserOptions": {
    "ecmaFeatures": {
      "jsx": true
    },
    "sourceType": "module"
  }
}

```

**Comment by nickjanssen** [comment]
Actually I've encountered more strange behaviour, unless I'm missing something:
![eslintbug2](https://cloud.githubusercontent.com/assets/2780160/16510883/0d47e8de-3f85-11e6-855d-41ef148f26cb.gif)

**Comment by ljharb** [comment]
I believe lowercased components are just treated as HTML element tag strings - ie, no error is expected there.

**Comment by lencioni** [comment]
This is happening because `Text` is a global in the browser environment.

Here's the data that ESLint uses to set up the globals for the environments: https://developer.mozilla.org/en-US/docs/Web/API/Text

And here's more information on the Text API: https://developer.mozilla.org/en-US/docs/Web/API/Text

**Comment by yannickcr** [comment]
Yep, It's an edge case an there is nothing we can do to fix this.
